ocaml/testsuite/tests/typing-layouts/error_message_attr.ml

```diff
@@ -0,0 +1,118 @@
+(* TEST
+   flags = "-extension layouts_alpha"
+ * expect
+*)
+
+module Float_u = Stdlib__Float_u
+
+[%%expect{|
+module Float_u = Stdlib__Float_u
+|}]
+
+(* Ltyp_var { name = None; layout } case *)
+let f (v: float#): ((_ : value)[@error_message "Custom message"]) = v
+[%%expect{|
+Line 1, characters 68-69:
+1 | let f (v: float#): ((_ : value)[@error_message "Custom message"]) = v
+                                                                        ^
+Error: This expression has type float# but an expression was expected of type
+         ('a : value)
+       The layout of float# is float64, because
+         it is the primitive float64 type float#.
+       But the layout of float# must be a sublayout of value, because
+         the [@error_message] attribute: Custom message.
```

Comment:
Hmm.... this isn't exactly right. It's note the `[@error_message]` that's causing the result to be a `value`.

Maybe something like

```
  of a layout constraint at line 1, columns xyz.
  Custom message
```

This makes it so that the constraint is what forces the layout, and just appends `Custom message` below. That seems to give the most flexibility to custom message authors.